Code.gs

```diff
@@ -271,57 +304,60 @@ function ConvertToCustomEvent(vevent){
   return event;
 }
 
-function ParseOrganizerName(veventString){
-  /*A regex match is necessary here because ICAL.js doesn't let us directly
-  * get the "CN" part of an ORGANIZER property. With something like
-  * ORGANIZER;CN="Sally Example":mailto:sally@example.com
-  * VEVENT.getFirstPropertyValue('organizer') returns "mailto:sally@example.com".
-  * Therefore we have to use a regex match on the VEVENT string instead
-  */
-
-  var nameMatch = RegExp("ORGANIZER(?:;|:)CN=(.*?):", "g").exec(veventString);
-  if (nameMatch != null && nameMatch.length > 1)
-    return nameMatch[1];
-  else
-    return null;
-}
-
-function ParseNotificationTime(notificationString){
-  //https://www.kanzaki.com/docs/ical/duration-t.html
-  var reminderTime = 0;
-
-  //We will assume all notifications are BEFORE the event
-  if (notificationString[0] == "+" || notificationString[0] == "-")
-    notificationString = notificationString.substr(1);
-
-  notificationString = notificationString.substr(1); //Remove "P" character
-
-  var secondMatch = RegExp("\\d+S", "g").exec(notificationString);
-  var minuteMatch = RegExp("\\d+M", "g").exec(notificationString);
-  var hourMatch = RegExp("\\d+H", "g").exec(notificationString);
-  var dayMatch = RegExp("\\d+D", "g").exec(notificationString);
-  var weekMatch = RegExp("\\d+W", "g").exec(notificationString);
-
-  if (weekMatch != null){
-    reminderTime += parseInt(weekMatch[0].slice(0, -1)) & 7 * 24 * 60 * 60; //Remove the "W" off the end
-
-    return reminderTime; //Return the notification time in seconds
+function CreateEvent(targetCalendar, customEvent){
+  var resultEvent;
+  
+  if (customEvent.isAllDay){
+    if (customEvent.recurrence != null){
+      //There is no "createAllDayEventSeries" that takes both a startTime and endTime as
+      //parameters. So instead we will create an all day event with these parameters and
+      //turn it into a series with a recurrence
+    
+      resultEvent = targetCalendar.createAllDayEvent(customEvent.title, 
+                                                     customEvent.startTime,
+                                                     customEvent.endTime,
+                                                     {
+                                                       location : customEvent.location, 
+                                                       description : customEvent.description
+                                                     });
+                                                     
+      resultEvent = resultEvent.getEventSeries().setRecurrence(customEvent.recurrence, 
```

Comment:
There's no need to set the start and end time for the EventSeries. Using setRecurrence(event, startDate) works just fine and will not turn AllDay Events to normal Events.